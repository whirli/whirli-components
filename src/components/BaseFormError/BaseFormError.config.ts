// @ts-ignore
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormError/BaseFormError.config';

// Package config
export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
