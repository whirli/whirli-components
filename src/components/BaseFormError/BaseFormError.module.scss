@import '@whirli-components/styles/Breakpoints.module.scss';

// default styles
.error {}
// breakpoint styles
$packageBreakpointStyles: ();