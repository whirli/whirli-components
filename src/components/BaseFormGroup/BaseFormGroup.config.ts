// @ts-ignore
import type { PropType } from '@composition';

import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormGroup/BaseFormGroup.config';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
