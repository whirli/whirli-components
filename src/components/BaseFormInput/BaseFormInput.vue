<template>
  <div :class="classes">Hello from BaseFormInput component</div>
</template>

<script setup lang="ts">
// Vue
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormInput/BaseFormInput.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormInput.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes = makeClasses(ComponentStyles, props, styles);
</script>
