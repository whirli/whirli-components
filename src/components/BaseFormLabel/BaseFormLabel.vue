<template>
  <label :for="props.name" :class="[classes, requiredClass]"><slot /></label>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormLabel/BaseFormLabel.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormLabel.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes: string[] = [styles.label, ...makeClasses(ComponentStyles, props, styles)];
const requiredClass: string = props.required ? styles.required : '';
</script>
