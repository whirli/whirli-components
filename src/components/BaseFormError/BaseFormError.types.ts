import { PropKeys, PropValues } from './BaseFormError.constants';
