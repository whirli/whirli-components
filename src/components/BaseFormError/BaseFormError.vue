<template>
  <BaseText v-bind="$attrs" :class="classes" v-if="$slots.default"><slot /></BaseText>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormError/BaseFormError.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormError.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

// @ts-ignore
const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
import BaseText from '@whirli-components/components/BaseText/BaseText.vue';
const { makeClasses } = useClasses();
const classes = [styles.error, ...makeClasses(ComponentStyles, ConfigProps, props, styles)];
</script>
