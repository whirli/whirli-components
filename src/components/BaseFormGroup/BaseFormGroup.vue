<template>
  <div :class="classes"><slot /></div>
</template>

<script setup lang="ts">
// Vue
;

// Styles
import styles from '@whirli-local/components/BaseFormGroup/BaseFormGroup.module.scss';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormGroup.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes: string[] = [styles['form-group'], ...makeClasses(ComponentStyles, props, styles)];
</script>
