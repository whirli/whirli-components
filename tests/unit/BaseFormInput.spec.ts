import { mount } from '@vue/test-utils';
import BaseFormInput from '@/components/BaseFormInput/BaseFormInput.vue';
import BaseFormGroup from '@/components/BaseFormGroup/BaseFormGroup.vue';
import BaseFormLabel from '@/components/BaseFormLabel/BaseFormLabel.vue';
import BaseFormError from '@/components/BaseFormError/BaseFormError.vue';

let wrapper: any;
const label = 'Hello World';
const content = 'Hello Content';
const error = 'Hello Error';
const name = 'select-name';

describe('BaseFormInput.vue', () => {
  beforeEach(() => {
    wrapper = mount(BaseFormInput, {
      slots: {
        default: content,
      },
      props: {
        label,
        error,
        name,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct label text', () => {
    expect(wrapper.findComponent(BaseFormLabel).text()).toMatch(label);
  });

  it('displays the correct error text', () => {
    expect(wrapper.findComponent(BaseFormError).text()).toMatch(error);
  });

  it('select has the correct name', () => {
    expect(wrapper.find('select').element.getAttribute('name')).toBe(name);
  });
});
