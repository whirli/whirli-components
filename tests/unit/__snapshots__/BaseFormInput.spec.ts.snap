// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormInput.vue renders the component 1`] = `
<div
  class=""
  disabled="false"
  multiple="false"
  type="text"
>
  
  <div
    class=""
  >
    
    <div
      class=""
    >
      
      
      <label
        class=""
        for="input-name"
      >
        
        Hello World
        
      </label>
      
      
    </div>
    <div
      class=""
    >
      
      
      
      <input
        aria-disabled="false"
        class=""
        id="input-name"
        name="input-name"
        type="text"
      />
      
      
      
    </div>
    <div
      class=""
    >
      
      
      <span
        class=""
      >
        
        
        Hello Error
        
        
      </span>
      
      
    </div>
    
  </div>
  
</div>
`;
